src/config/parser.rs

```diff
@@ -42,6 +42,8 @@ pub struct Config {
     /// It stores the level of safe search to be used for restricting content in the
     /// search results.
     pub safe_search: u8,
+    /// It stores the TCP connection keepalive duration in seconds.
+    pub tcp_connection_keepalive: u64,
```

Comment:
```suggestion
    pub tcp_connection_keepalive: u8,
```

Finally, I would like to suggest one last minor change, I think instead of parsing and the storing the `tcp_connection_keepalive` option as u64. I feel it would be better to parse it with a smaller data type like `u8`. Because using a larger data type can allow the user to mistakenly put a huge value like `10000000` which is not great as you know like keeping TCP alive `100000000` seconds is dangerous as it can overly consume the system resources. So instead using a smaller data type like `u8` would be better as it would only allow the user to provide a maximum value for this option to `256 seconds. :slightly_smiling_face: